src/devices/machine/e05a30.cpp

```diff
@@ -167,6 +168,13 @@ void e05a30_device::write(offs_t offset, uint8_t data)
 	LOG("%s: e05a30_w([0xC0%02x]): %02x\n", machine().describe_context(), offset, data);
 
 	switch (offset) {
+	case 0x0:
+	case 0x1:
+	case 0x2:
+	case 0x3:
+		m_c000_shift_register &= ~(0xff << ((3-offset)*8));
+		m_c000_shift_register |= (data << ((3-offset)*8));
```

Comment:
You’re relying on integer promotion rules here – please make it explicit and cast `0xff` and `data` to the desired type.